F-Droid debug builds crash on startup

### Steps to reproduce

1. download and install the vector-fdroid-arm64-v8a-debug.apk from [Build #10739](https://buildkite.com/matrix-dot-org/element-android/builds/10739) or newer
2. try to start Element dbg

I've also tried it on a Oneplus 6T that has Play services installed (my Oneplus 5T doesn't), same result. The GPlay builds itself seem fine though.

I suspect it's related to #6720, as that's the PR of the first build that doesn't work anymore on my devices.

### Outcome

#### What did you expect?
That the app starts and I can use it normally

#### What happened instead?
It crashed while loading the app, I can't even see the loading animation/Element logo properly.

### Your phone model

Oneplus 5T

### Operating system version

Android 12/LineageOS 19-20220830-NIGHTLY-dumpling

### Application version and app store

vector-fdroid-arm64-v8a-debug.apk from [Build #10739](https://buildkite.com/matrix-dot-org/element-android/builds/10739)

### Homeserver

I'm not even logged in at that point

### Will you send logs?

No

### Are you willing to provide a PR?

No

Thanks for the report, I reproduced the crash. The stacktrace is not relevant but it is probably caused by the modularisation as you suggested.

Stacktrace:
```2022-09-01 19:25:37.084 im.vector.app.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: im.vector.app.debug, PID: 4187
    java.lang.RuntimeException: Unable to start activity ComponentInfo{im.vector.app.debug/im.vector.app.features.MainActivity}: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3685)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3842)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at dagger.hilt.EntryPoints.get(EntryPoints.java:62)
        at im.vector.app.core.extensions.ContextKt.singletonEntryPoint(Context.kt:45)
        at im.vector.app.core.platform.VectorBaseActivity.onCreate(VectorBaseActivity.kt:205)
        at im.vector.app.features.MainActivity.onCreate(MainActivity.kt:143)
        at android.app.Activity.performCreate(Activity.java:8054)
        at android.app.Activity.performCreate(Activity.java:8034)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3666)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3842) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7842) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
2022-09-01 19:25:37.086 ? E/JitsiMeetSDK: JitsiMeetUncaughtExceptionHandler FATAL ERROR
    java.lang.RuntimeException: Unable to start activity ComponentInfo{im.vector.app.debug/im.vector.app.features.MainActivity}: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3685)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3842)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at dagger.hilt.EntryPoints.get(EntryPoints.java:62)
        at im.vector.app.core.extensions.ContextKt.singletonEntryPoint(Context.kt:45)
        at im.vector.app.core.platform.VectorBaseActivity.onCreate(VectorBaseActivity.kt:205)
        at im.vector.app.features.MainActivity.onCreate(MainActivity.kt:143)
        at android.app.Activity.performCreate(Activity.java:8054)
        at android.app.Activity.performCreate(Activity.java:8034)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3666)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3842) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7842) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
    
    java.lang.RuntimeException: Unable to start activity ComponentInfo{im.vector.app.debug/im.vector.app.features.MainActivity}: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3685)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3842)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2252)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.IllegalStateException: Given component holder class android.app.Application does not implement interface dagger.hilt.internal.GeneratedComponent or interface dagger.hilt.internal.GeneratedComponentManager
        at dagger.hilt.EntryPoints.get(EntryPoints.java:62)
        at im.vector.app.core.extensions.ContextKt.singletonEntryPoint(Context.kt:45)
        at im.vector.app.core.platform.VectorBaseActivity.onCreate(VectorBaseActivity.kt:205)
        at im.vector.app.features.MainActivity.onCreate(MainActivity.kt:143)
        at android.app.Activity.performCreate(Activity.java:8054)
        at android.app.Activity.performCreate(Activity.java:8034)
2022-09-01 19:25:37.086 ? E/JitsiMeetSDK:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3666)
        	... 12 more```

--- a/vector-app/build.gradle
+++ b/vector-app/build.gradle
@@ -340,10 +340,6 @@ android {
                 "-opt-in=kotlinx.coroutines.ExperimentalCoroutinesApi",
         ]
     }
-
-    buildFeatures {
-        viewBinding true
-    }
 }
 
 dependencies {

--- a/vector-app/src/gplay/AndroidManifest.xml
+++ b/vector-app/src/gplay/AndroidManifest.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
-    <application android:name="im.vector.app.VectorApplication">
+    <application>
 
         <!-- Firebase components -->
         <meta-data

--- a/vector-app/src/main/AndroidManifest.xml
+++ b/vector-app/src/main/AndroidManifest.xml
@@ -4,6 +4,7 @@
     package="im.vector.application">
 
     <application
+        android:name="im.vector.app.VectorApplication"
         android:allowBackup="false"
         android:hasFragileUserData="true"
         android:icon="@mipmap/ic_launcher"

--- a/vector/build.gradle
+++ b/vector/build.gradle
@@ -78,17 +78,10 @@ android {
     productFlavors {
         gplay {
             dimension "store"
-            isDefault = true
-
-            buildConfigField "String", "SHORT_FLAVOR_DESCRIPTION", "\"G\""
-            buildConfigField "String", "FLAVOR_DESCRIPTION", "\"GooglePlay\""
         }
 
         fdroid {
             dimension "store"
-
-            buildConfigField "String", "SHORT_FLAVOR_DESCRIPTION", "\"F\""
-            buildConfigField "String", "FLAVOR_DESCRIPTION", "\"FDroid\""
         }
     }
 

